stemcell_builder/stages/image_create_efi/apply.sh

```
@@ -9,10 +9,21 @@ disk_image=${work}/${stemcell_image_name}
 dd if=/dev/null of=${disk_image} bs=1M seek=${image_create_disk_size} 2> /dev/null
-parted --script ${disk_image} mklabel msdos
-parted --script ${disk_image} mkpart primary fat32 0% 49MiB
-parted --script ${disk_image} set 1 esp on
-parted --script ${disk_image} mkpart primary ext2 50MiB 100%
+
+# Partition the disk image using sfdisk
+# sfdisk works better than parted in virtualized environments (Docker Desktop on Apple Silicon)
+# Partition layout:
+#   - Partition 1: EFI System Partition (ESP), ~49MiB, type EF (EFI)
+#   - Partition 2: Linux root partition, remaining space, type 83 (Linux)
+sfdisk ${disk_image} <<EOF
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)